mapper/tedge_mapper/src/sm_c8y_mapper/error.rs

```diff
@@ -23,20 +17,27 @@ pub(crate) enum SmartRestSerializerError {
 
 #[derive(thiserror::Error, Debug)]
 pub(crate) enum SmartRestDeserializerError {
-    #[error("The SmartREST message ID is not Update Software Operation (528).")]
-    NotUpdateSoftwareOperation,
+    #[error("The received SmartREST message ID {id} is unsupported.")]
+    UnsupportedOperation { id: String },
 
     #[error("Failed to deserialize SmartREST.")]
     InvalidCsv(#[from] csv::Error),
 
-    #[error(
-        "Action {action} is not recognized. Acceptable software actions are install or delete."
-    )]
+    #[error("Action {action} is not recognized. It must be install or delete.")]
     ActionNotFound { action: String },
 }
 
+#[derive(thiserror::Error, Debug)]
+pub(crate) enum MapperTopicError {
+    #[error("Topic {topic} is unknown.")]
+    UnknownTopic { topic: String },
+}
+
 #[derive(thiserror::Error, Debug)]
 pub(crate) enum SMCumulocityMapperError {
+    #[error(transparent)]
+    InvalidTopicError(#[from] MapperTopicError),
```

Comment:
Didn't really understand why this enum redirection was done. Why not define that `UnknownTopic` enum variant here directly?

Reply:
I wanted to group errors per file. Unknown topic error comes from the file `topic.rs`. and SMCumulocityMapperError comes from the file `mapper.rs`.